crates/ruff/src/rules/pycodestyle/rules/tab_indentation.rs

```diff
@@ -16,15 +16,48 @@ impl Violation for TabIndentation {
 }
 
 /// W191
-pub fn tab_indentation(lineno: usize, line: &str) -> Option<Diagnostic> {
+pub fn tab_indentation(lineno: usize, line: &str, string_ranges: &[Range]) -> Option<Diagnostic> {
     let indent = leading_space(line);
+    if let Some(tab_index) = indent.find('\t') {
+        // If the tab character is within a multi-line string, abort.
+        if let Ok(range_index) = string_ranges.binary_search_by(|range| {
+            let start = range.location.row();
+            let end = range.end_location.row();
+            if start > lineno {
+                std::cmp::Ordering::Greater
+            } else if end < lineno {
+                std::cmp::Ordering::Less
+            } else {
+                std::cmp::Ordering::Equal
+            }
+        }) {
+            let string_range = &string_ranges[range_index];
+            let start = string_range.location;
+            let end = string_range.end_location;
+
+            // Tab is contained in the string range by virtue of lines.
+            if lineno != start.row() && lineno != end.row() {
+                return None;
+            }
+
+            let tab_column = line[..tab_index].chars().count();
```

Comment:
(I think we need to count characters here, since `.column()` is based on characters.)

Reply:
How do you mean?

Reply:
`indent.find('\t')` returns the index of the starting _byte_ for the tab character, but some characters can require multiple bytes (think unicode) -- so `str.len()` in Rust doesn't always return what you might expect if coming from another programming language.

Reply:
Oh interesting, I didn't know that. I'll have to read up on it, thanks!